aten/src/ATen/native/cpu/group_norm_kernel.cpp

```diff
@@ -316,16 +417,16 @@ void GroupNormKernelImplChannelsLastInternal(
         T_ACC mean_val{0}, rstd_val{0};
         for (const auto d : c10::irange(D)) {
           for (const auto t : c10::irange(num_threads)) {
-            T* buffer_ptr = buffer_data + t * N * 2 * C + n * 2 * C;
+            T_ACC* buffer_ptr = buffer_data + t * N * 2 * C + n * 2 * C;
             mean_val += buffer_ptr[g * D + d];
             rstd_val += buffer_ptr[g * D + d + C];
            }
         }
         mean_val *= s;
         rstd_val = std::max(rstd_val * s - mean_val * mean_val, T_ACC(0));
         rstd_val = T_ACC(1) / std::sqrt(rstd_val + eps);
-        mean_data[n * G + g] = T(mean_val);
-        rstd_data[n * G + g] = T(rstd_val);
+        tmp_buffer_data[n * 2 * G + 2 * g] = mean_val;
+        tmp_buffer_data[n * 2 * G + 2 * g + 1] = rstd_val;
```

Comment:
May I know the reason why we don't store the mean/rstd to mean_data and rstd_data directly but use a temp buffer to do it?

Reply:
Because if dtype of mean or rstd is bfloat16, it will loss precision.